default/AI/ShipDesignAI.py

```diff
@@ -1445,7 +1404,7 @@ def _rating_function(self):
         if self.troops == 0:
             return INVALID_DESIGN_RATING
         else:
-            return self.troops/(self.production_cost**((1+foAI.foAIstate.shipCount * AIDependencies.SHIP_UPKEEP)**-1))
+            return self.troops/self._adjusted_production_cost()
```

Comment:
Adding a whitespace around ```/```feels entirely unnatural for me. I would not use ```a ** b``` over ```a**b``` either except for very rare occasions. I consider these to be a translation from actual math to code and thus adding spaces where there wouldn't be spaces in a written equation makes it harder for me to immediately grab the concept of the math and thus reduces readability of the code.
The same mostly goes for multiplication as well. There are very rare cases where I explicitly denote multiplication but mostly, in real life ```a * b``` is written as ```ab``` and thus ```a*b``` is the natural conversion into code for me.
As said, there are exceptions for very long variable names which make it hard to see where one starts and one ends for example.

I do not think PEP8 conventions contradict this interpretation. In fact, the guideline to add whitespaces around low-priority operators (and thus group higher priority operators together) is a direct consequence from my approach.

So unless there is a very clear rule on this which seems not to be the case, I will go with what is most readable and natural for me.